src/main/java/ch/njol/skript/conditions/CondHasItemCooldown.java

```diff
@@ -55,22 +59,23 @@ public class CondHasItemCooldown extends Condition {
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		players = (Expression<Player>) exprs[0];
 		itemtypes = (Expression<ItemType>) exprs[1];
-		setNegated(matchedPattern == 1);
+		setNegated(matchedPattern > 1);
 		return true;
 	}
 
 	@Override
-	public boolean check(Event e) {
-		return players.check(e, (p) -> 
-				itemtypes.check(e, (it) -> 
-						p.hasCooldown(it.getMaterial())
+	public boolean check(Event event) {
+		return players.check(event, (player) ->
+				itemtypes.check(event, (itemType) ->
+						itemType.hasType() && player.hasCooldown(itemType.getMaterial())
```

Comment:
Same thing as last review:
![image](https://user-images.githubusercontent.com/29547183/219855225-6d137962-af88-495d-94c8-ffda2a4d7072.png)
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/Material.html#isItem()

Reply:
It should be covered by `ItemType#hasType()`, if I am not mistaken, materials that are not items don't have ItemData

Reply:
> It should be covered by `ItemType#hasType()`, if I am not mistaken, materials that are not items don't have ItemData

form my quick search didn't see `ItemType#hasType()` but found `ItemType#hasItem` which seems to do what you suggested just seems poorly worded

Reply:
> From my quick search didn't see ItemType#hasType()

It has been [added](https://github.com/SkriptLang/Skript/pull/4198/files/84ed9e07409b4c3583a12b44f5e6c3b12b8942f2#diff-1578d77b4b4b435cfe11847dbe3451001a3ef28efca17d2ea12896cd5f36a5a7R352) in this PR

Reply:
> It should be covered by `ItemType#hasType()`, if I am not mistaken, materials that are not items don't have ItemData

Also fine, but make sure that is documented on the method